pyrobosim/pyrobosim/utils/pose.py

```diff
@@ -199,6 +199,34 @@ def __repr__(self):
         )
         return f"Pose: [{pos_str}, {quat_str}]"
 
+    def is_approx(self, other, rel_tol=1e-09, abs_tol=0.0):
+        if not (isinstance(other, Pose)):
+            raise TypeError("Expected a Pose")
+
+        a = np.array([self.x, self.y, self.z])
+        b = np.array([other.x, other.y, other.z])
+
+        # rel_tol = 1e-09 # setting rel_tol here makes it a float
+
+        # Just for debugging - rel_tol is a Pose for some reason?
+        print(
+            isinstance(a[0], float),
+            isinstance(a[1], float),
+            isinstance(a[2], float),
+            isinstance(b[0], float),
+            isinstance(b[1], float),
+            isinstance(b[2], float),
+            isinstance(rel_tol, float),
+            isinstance(abs_tol, float),
+        )
+
+        return np.allclose(a, b, rel_tol, abs_tol) and nearly_equivalent(
+            self.q, other.q
+        )  # default rtol=1e-05, atol=1e-08 for quaternion
+
+    def __eq__(self, other):
+        return self.is_approx(self, other)
```

Comment:
rel_tol is appearing to be a pose because the `self` shouldn't be explicitly passed in as an argument.

Should be e.g.

```
self.is_approx(other)
self.is_approx(other, rel_tol=1e-6, abs_tol=1e-4)

Reply:
Also I think the `__eq__()` method should check actual equality, whereas `is_approx()` can be called separately.

Reply:
Ah yes, thanks - changed __eq__() as well